kotsadm/pkg/handlers/identity.go

```diff
@@ -111,68 +80,183 @@ func ConfigureIdentityService(w http.ResponseWriter, r *http.Request) {
 		JSON(w, http.StatusInternalServerError, response)
 		return
 	}
 
 	cfg, err := config.GetConfig()
 	if err != nil {
 		logger.Error(err)
-		response.Error = "failed to get cluster config"
+		response := ErrorResponse{Error: "failed to get cluster config"}
 		JSON(w, http.StatusInternalServerError, response)
 		return
 	}
 
 	clientset, err := kubernetes.NewForConfig(cfg)
 	if err != nil {
 		logger.Error(err)
-		response.Error = "failed to create kubernetes clientset"
+		response := ErrorResponse{Error: "failed to create kubernetes clientset"}
 		JSON(w, http.StatusInternalServerError, response)
 		return
 	}
 
 	registryOptions, err := kotsadm.GetKotsadmOptionsFromCluster(namespace, clientset)
 	if err != nil {
 		logger.Error(err)
-		response.Error = "failed to get kotsadm options from cluster"
+		response := ErrorResponse{Error: "failed to get kotsadm options from cluster"}
 		JSON(w, http.StatusInternalServerError, response)
 		return
 	}
 
 	if err := identity.Deploy(r.Context(), clientset, namespace, identityConfig, *ingressConfig, &registryOptions); err != nil {
 		logger.Error(err)
-		response.Error = "failed to deploy the identity service"
+		response := ErrorResponse{Error: "failed to deploy the identity service"}
 		JSON(w, http.StatusInternalServerError, response)
 		return
 	}
 
-	response.Success = true
+	response := ErrorResponse{
+		Success: true,
+	}
 
 	JSON(w, http.StatusOK, response)
 }
 
+type DexConnectorInfo struct {
+	Type   string
+	ID     string
+	Name   string
+	Config []byte
+}
+
+func getDexConnectorInfo(request ConfigureIdentityServiceRequest) (*DexConnectorInfo, error) {
+	var connectorConfig interface{}
+	dexConnectorInfo := DexConnectorInfo{}
+
+	if request.OIDCConfig != nil {
+		c := oidc.Config{
+			Issuer:                    request.OIDCConfig.Issuer,
+			ClientID:                  request.OIDCConfig.ClientID,
+			ClientSecret:              request.OIDCConfig.ClientSecret,
+			RedirectURI:               "{{OIDCIdentityCallbackURL}}",
+			GetUserInfo:               true,
+			UserNameKey:               "email",
+			InsecureSkipEmailVerified: false,
+			InsecureEnableGroups:      true,
+			Scopes: []string{
+				"openid",
+				"email",
+				"groups",
+			},
+		}
+
+		if request.OIDCConfig.GetUserInfo != nil {
+			c.GetUserInfo = *request.OIDCConfig.GetUserInfo
+		}
+		if request.OIDCConfig.UserNameKey != "" {
+			c.UserNameKey = request.OIDCConfig.UserNameKey
+		}
+		if request.OIDCConfig.InsecureSkipEmailVerified != nil {
+			c.InsecureSkipEmailVerified = *request.OIDCConfig.InsecureSkipEmailVerified
+		}
+		if request.OIDCConfig.InsecureEnableGroups != nil {
+			c.InsecureEnableGroups = *request.OIDCConfig.InsecureEnableGroups
+		}
+		if len(request.OIDCConfig.Scopes) > 0 {
+			c.Scopes = append(c.Scopes, request.OIDCConfig.Scopes...)
+		}
+
+		dexConnectorInfo.Type = "oidc"
+		dexConnectorInfo.ID = "openid"
+		dexConnectorInfo.Name = "OpenID"
```

Comment:
i dont think you should hardcode id or name, for example im using okta but i dont want it to say "openid"